mlflow/resource_mlflow_model.go

```diff
@@ -3,112 +3,78 @@ package mlflow
-	CreationTimestamp    int64          `json:"creation_timestamp,omitempty" tf:"computed"`
-	LastUpdatedTimestamp int64          `json:"last_updated_timestamp,omitempty" tf:"computed"`
-	UserID               string         `json:"user_id,omitempty" tf:"computed"`
-	LatestVersions       []ModelVersion `json:"latest_versions,omitempty" tf:"computed"`
-	Description          string         `json:"description,omitempty"`
-	Tags                 []Tag          `json:"tags,omitempty"`
-	RegisteredModelID    string         `json:"id,omitempty" tf:"computed,alias:registered_model_id"`
-}
-
-// registeredModel defines response from GET API op
-type registeredModel struct {
-	RegisteredModelDatabricks Model `json:"registered_model_databricks"`
-}
-
-type ModelsAPI struct {
-	client  *common.DatabricksClient
-	context context.Context
-}
-
-func NewModelsAPI(ctx context.Context, m any) ModelsAPI {
-	return ModelsAPI{m.(*common.DatabricksClient), ctx}
-}
-
-func (a ModelsAPI) Create(m *Model) error {
-	return a.client.Post(a.context, "/mlflow/registered-models/create", m, m)
-}
-
-func (a ModelsAPI) Read(name string) (*Model, error) {
-	var m registeredModel
-	err := a.client.Get(a.context, "/mlflow/databricks/registered-models/get", map[string]string{
-		"name": name,
-	}, &m)
-	if err != nil {
-		return nil, err
-	}
-	return &m.RegisteredModelDatabricks, nil
-}
-
-// Update the model entity
-func (a ModelsAPI) Update(m *Model) error {
-	return a.client.Patch(a.context, "/mlflow/registered-models/update", m)
-}
-
-// Delete removes the model by its name
-func (a ModelsAPI) Delete(name string) error {
-	return a.client.Delete(a.context, "/mlflow/registered-models/delete", map[string]string{
-		"name": name,
-	})
-}
-
 func ResourceMlflowModel() *schema.Resource {
 	s := common.StructToSchema(
-		Model{},
-		func(m map[string]*schema.Schema) map[string]*schema.Schema {
-			delete(m, "latest_versions")
-			return m
+		ml.Model{},
+		func(s map[string]*schema.Schema) map[string]*schema.Schema {
+			delete(s, "latest_versions")
+			s["name"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			}
+			return s
 		})
 
 	return common.Resource{
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			var m Model
-			common.DataToStructPointer(d, s, &m)
-			if err := NewModelsAPI(ctx, c).Create(&m); err != nil {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var req ml.CreateModelRequest
+			common.DataToStructPointer(d, s, &req)
+			res, err := w.ModelRegistry.CreateModel(ctx, req)
+			if err != nil {
+				return err
+			}
+			model, err := w.ModelRegistry.GetModel(ctx, ml.GetModelRequest{
+				Name: res.RegisteredModel.Name,
+			})
+			if err != nil {
 				return err
 			}
-			d.SetId(m.Name)
+			d.SetId(model.RegisteredModelDatabricks.Name)
+			d.Set("registered_model_id", model.RegisteredModelDatabricks.Id) // alias
```

Comment:
Is this a bugfix? Is it not already handled in the `Read` implementation below?

Reply:
The `RegisteredModel` object that is returned by the CREATE request does not have an ID in the payload. To get the model ID we have to run a GET. Not sure why it's designed like this but I'll ask

Reply:
Yeah, I saw this as well. I just missed how this worked in the original implementation :p